dali/kernels/imgproc/paste/paste_gpu.h

```diff
@@ -0,0 +1,293 @@
+        }
+      }
+  }
+}
+
+
+
+template <class OutputType, class InputType, int ndims>
+__global__ void
+PasteKernel(const SampleDescriptorGPU<OutputType, InputType, ndims> *samples,
+                  const PatchDesc<InputType, ndims> *patchs,
+                  const BlockDesc<ndims> *blocks) {
+  static_assert(ndims == 2, "Function requires 2 dimensions in the input");
+  const auto &block = blocks[blockIdx.x];
+  const auto &sample = samples[block.sample_idx];
+  const PatchDesc<InputType, ndims> *my_patchs = patchs + sample.patch_start_idx;
+
+
+  auto *__restrict__ out = sample.out;
+
+  int patch_y = 0;
+  int min_patch_x = 0;
+  while (threadIdx.x + block.start.x >= my_patchs[min_patch_x].patch_end[1]) min_patch_x++;
+
+  for (int y = threadIdx.y + block.start.y; y < block.end.y; y += blockDim.y) {
+    while (y >= my_patchs[patch_y * sample.patch_counts[1]].patch_end[0]) patch_y++;
+    int patch_x = min_patch_x;
+    for (int x = threadIdx.x + block.start.x; x < block.end.x; x += blockDim.x) {
+      while (x >= my_patchs[patch_y * sample.patch_counts[1] + patch_x].patch_end[1]) patch_x++;
+      const PatchDesc<InputType, ndims> *patch =
+              &my_patchs[patch_y * sample.patch_counts[1] + patch_x];
+      if (patch->in == nullptr) {
+          out[y * sample.out_pitch[1] + x] = 0;
+      } else {
+        out[y * sample.out_pitch[1] + x] = ConvertSat<OutputType>(
+                patch->in[(y - patch->patch_start[0] + patch->in_anchor[0]) * patch->in_pitch[1]
+                         + (x - patch->patch_start[1] + patch->in_anchor[1])]);
+      }
+    }
+  }
+}
+
+}  // namespace paste
+
+template <typename OutputType, typename InputType, int ndims>
+class PasteGPU {
+ private:
+  static constexpr size_t spatial_dims = ndims - 1;
+  using BlockDesc = kernels::BlockDesc<spatial_dims>;
+  using SampleDesc = paste::SampleDescriptorGPU<OutputType, InputType, spatial_dims>;
+  using PatchDesc = paste::PatchDesc<InputType, spatial_dims>;
+
+  std::vector<SampleDesc> sample_descriptors_;
+  std::vector<PatchDesc> patch_descriptors_;
+
+ public:
+  BlockSetup<spatial_dims, -1 /* No channel dimension, only spatial */> block_setup_;
+
+  KernelRequirements Setup(
+      KernelContext &context,
+      const InListGPU<InputType, ndims> &in,
+      span<paste::MultiPasteSampleInput<spatial_dims>> samples,
+      TensorListShape<ndims> out_shape) {
+    paste::CreateSampleDescriptors(sample_descriptors_, patch_descriptors_, in, samples);
+
+    KernelRequirements req;
+    ScratchpadEstimator se;
+    auto flattened_shape = collapse_dim(out_shape, 1);
+    block_setup_.SetupBlocks(flattened_shape, true);
+    se.add<SampleDesc>(AllocType::GPU, sample_descriptors_.size());
+    se.add<PatchDesc>(AllocType::GPU, patch_descriptors_.size());
+    se.add<BlockDesc>(AllocType::GPU, block_setup_.Blocks().size());
+    // req.output_shapes = {in.shape}; Once again, this is determined by operator
```

Comment:
If we still use KernelRequirements, I guess we should output the shape - even though here it doesn't make much sense (and it's quite heavy).
```suggestion
    req.output_shapes = { out_shape };
```